### PR TITLE
Add helm NOTES to the root folder

### LINKED ISSUE #10070: Add helm NOTES file
Helm supports adding a custom piece of text after install, to give users some first pointers and advice what to do next. We should add this with the release, name, maybe a pointer to Slack.

https://helm.sh/docs/chart_template_guide/notes_files/

### PR DESCRIPTION
**Pull request by soumynathan**
This patch adds helm NOTES file to the root folder.

Fixes: #10070
Signed-off-by: Swaminathan Vasudevan <svasudevan@suse.com>

### CODE REVIEW

**Comment by soumynathan** [comment]
@joestringer  This is what the output looks like.
```
master-node:~/go/src/github.com/cilium/install/kubernetes$ helm install --debug --dry-run --namespace kube-system -f ./cilium/values.yaml cilium ./cilium
install.go:158: [debug] Original chart version: ""
install.go:175: [debug] CHART PATH: /home/ubuntu/go/src/github.com/cilium/install/kubernetes/cilium

NAME: cilium
LAST DEPLOYED: Wed Apr  8 19:35:45 2020
NAMESPACE: kube-system
STATUS: pending-install
REVISION: 1
TEST SUITE: None
USER-SUPPLIED VALUES:
agent:
  enabled: true
  keepDeprecatedLabels: false
  sleepAfterInit: false
config:
  enabled: true
global:
  autoDirectNodeRoutes: false
  azure:
    enabled: false
  bpf:
    ctAnyMax: 262144
    ctTcpMax: 524288
    monitorAggregation: medium
    monitorFlags: all
    monitorInterval: 5s
    natMax: 524288
    preallocateMaps: false
    waitForMount: false
  cleanBpfState: false
  cleanState: false
  cluster:
    name: default
  cni:
    binPath: /opt/cni/bin
    chainingMode: none
    confFileMountPath: /tmp/cni-configuration
    confPath: /etc/cni/net.d
    configMapKey: cni-config
    customConf: false
    hostConfDirMountPath: /host/etc/cni/net.d
    install: true
  containerRuntime:
    integration: none
  daemon:
    runPath: /var/run/cilium
  debug:
    enabled: false
  enableXTSocketFallback: true
  encryption:
    enabled: false
    keyFile: keys
    mountPath: /etc/ipsec
    nodeEncryption: false
    secretName: cilium-ipsec-keys
  endpointRoutes:
    enabled: false
  eni: false
  etcd:
    clusterDomain: cluster.local
    clusterSize: 3
    enabled: false
    endpoints:
    - https://CHANGE-ME:2379
    managed: false
    ssl: false
  externalIPs:
    enabled: false
  flannel:
    enabled: false
    masterDevice: cni0
    uninstallOnExit: false
  hostPort:
    enabled: false
  hostServices:
    enabled: false
    protocols: tcp,udp
  hubble:
    enabled: false
    eventQueueSize: null
    flowBufferSize: null
    listenAddresses: []
    metricServer: null
    metrics: []
    socketPath: /var/run/cilium/hubble.sock
  identityAllocationMode: crd
  identityChangeGracePeriod: 5s
  installIptablesRules: true
  ipv4:
    enabled: true
  ipv6:
    enabled: false
  ipvlan:
    enabled: false
  k8s:
    requireIPv4PodCIDR: false
  kubeProxyReplacement: probe
  logSystemLoad: false
  masquerade: true
  nodePort:
    acceleration: none
    autoProtectPortRange: true
    enabled: false
    mode: hybrid
  nodeinit:
    bootstrapFile: /tmp/cilium-bootstrap-time
    enabled: false
  operatorPrometheus:
    enabled: false
    port: 6942
  policyAuditMode: false
  pprof:
    enabled: false
  prometheus:
    enabled: false
    port: 9090
    serviceMonitor:
      enabled: false
  psp:
    enabled: false
  pullPolicy: Always
  registry: docker.io/cilium
  remoteNodeIdentity: true
  sockops:
    enabled: false
  synchronizeK8sNodes: true
  tag: latest
  tls:
    secretsBackend: local
  tunnel: vxlan
  wellKnownIdentities:
    enabled: false
operator:
  enabled: true
preflight:
  enabled: false
  tofqdnsPreCache: ""
  validateCNPs: true

COMPUTED VALUES:
agent:
  enabled: true
  global:
    autoDirectNodeRoutes: false
    azure:
      enabled: false
    bpf:
      ctAnyMax: 262144
      ctTcpMax: 524288
      monitorAggregation: medium
      monitorFlags: all
      monitorInterval: 5s
      natMax: 524288
      preallocateMaps: false
      waitForMount: false
    cleanBpfState: false
    cleanState: false
    cluster:
      name: default
    cni:
      binPath: /opt/cni/bin
      chainingMode: none
      confFileMountPath: /tmp/cni-configuration
      confPath: /etc/cni/net.d
      configMapKey: cni-config
      customConf: false
      hostConfDirMountPath: /host/etc/cni/net.d
      install: true
    containerRuntime:
      integration: none
    daemon:
      runPath: /var/run/cilium
    debug:
      enabled: false
    enableXTSocketFallback: true
    encryption:
      enabled: false
      keyFile: keys
      mountPath: /etc/ipsec
      nodeEncryption: false
      secretName: cilium-ipsec-keys
    endpointRoutes:
      enabled: false
    eni: false
    etcd:
      clusterDomain: cluster.local
      clusterSize: 3
      enabled: false
      endpoints:
      - https://CHANGE-ME:2379
      managed: false
      ssl: false
    externalIPs:
      enabled: false
    flannel:
      enabled: false
      masterDevice: cni0
      uninstallOnExit: false
    hostPort:
      enabled: false
    hostServices:
      enabled: false
      protocols: tcp,udp
    hubble:
      enabled: false
      eventQueueSize: null
      flowBufferSize: null
      listenAddresses: []
      metricServer: null
      metrics: []
      socketPath: /var/run/cilium/hubble.sock
    identityAllocationMode: crd
    identityChangeGracePeriod: 5s
    installIptablesRules: true
    ipv4:
      enabled: true
    ipv6:
      enabled: false
    ipvlan:
      enabled: false
    k8s:
      requireIPv4PodCIDR: false
    kubeProxyReplacement: probe
    logSystemLoad: false
    masquerade: true
    nodePort:
      acceleration: none
      autoProtectPortRange: true
      enabled: false
      mode: hybrid
    nodeinit:
      bootstrapFile: /tmp/cilium-bootstrap-time
      enabled: false
    operatorPrometheus:
      enabled: false
      port: 6942
    policyAuditMode: false
    pprof:
      enabled: false
    prometheus:
      enabled: false
      port: 9090
      serviceMonitor:
        enabled: false
    psp:
      enabled: false
    pullPolicy: Always
    registry: docker.io/cilium
    remoteNodeIdentity: true
    sockops:
      enabled: false
    synchronizeK8sNodes: true
    tag: latest
    tls:
      secretsBackend: local
    tunnel: vxlan
    wellKnownIdentities:
      enabled: false
  image: cilium
  initResources:
    requests:
      cpu: 100m
      memory: 100Mi
  keepDeprecatedLabels: false
  maxUnavailable: 2
  monitor:
    enabled: false
    eventTypes: []
    resources: {}
  resources: {}
  sleepAfterInit: false
config:
  enabled: true
  global:
    autoDirectNodeRoutes: false
    azure:
      enabled: false
    bpf:
      ctAnyMax: 262144
      ctTcpMax: 524288
      monitorAggregation: medium
      monitorFlags: all
      monitorInterval: 5s
      natMax: 524288
      preallocateMaps: false
      waitForMount: false
    cleanBpfState: false
    cleanState: false
    cluster:
      name: default
    cni:
      binPath: /opt/cni/bin
      chainingMode: none
      confFileMountPath: /tmp/cni-configuration
      confPath: /etc/cni/net.d
      configMapKey: cni-config
      customConf: false
      hostConfDirMountPath: /host/etc/cni/net.d
      install: true
    containerRuntime:
      integration: none
    daemon:
      runPath: /var/run/cilium
    debug:
      enabled: false
    enableXTSocketFallback: true
    encryption:
      enabled: false
      keyFile: keys
      mountPath: /etc/ipsec
      nodeEncryption: false
      secretName: cilium-ipsec-keys
    endpointRoutes:
      enabled: false
    eni: false
    etcd:
      clusterDomain: cluster.local
      clusterSize: 3
      enabled: false
      endpoints:
      - https://CHANGE-ME:2379
      managed: false
      ssl: false
    externalIPs:
      enabled: false
    flannel:
      enabled: false
      masterDevice: cni0
      uninstallOnExit: false
    hostPort:
      enabled: false
    hostServices:
      enabled: false
      protocols: tcp,udp
    hubble:
      enabled: false
      eventQueueSize: null
      flowBufferSize: null
      listenAddresses: []
      metricServer: null
      metrics: []
      socketPath: /var/run/cilium/hubble.sock
    identityAllocationMode: crd
    identityChangeGracePeriod: 5s
    installIptablesRules: true
    ipv4:
      enabled: true
    ipv6:
      enabled: false
    ipvlan:
      enabled: false
    k8s:
      requireIPv4PodCIDR: false
    kubeProxyReplacement: probe
    logSystemLoad: false
    masquerade: true
    nodePort:
      acceleration: none
      autoProtectPortRange: true
      enabled: false
      mode: hybrid
    nodeinit:
      bootstrapFile: /tmp/cilium-bootstrap-time
      enabled: false
    operatorPrometheus:
      enabled: false
      port: 6942
    policyAuditMode: false
    pprof:
      enabled: false
    prometheus:
      enabled: false
      port: 9090
      serviceMonitor:
        enabled: false
    psp:
      enabled: false
    pullPolicy: Always
    registry: docker.io/cilium
    remoteNodeIdentity: true
    sockops:
      enabled: false
    synchronizeK8sNodes: true
    tag: latest
    tls:
      secretsBackend: local
    tunnel: vxlan
    wellKnownIdentities:
      enabled: false
global:
  autoDirectNodeRoutes: false
  azure:
    enabled: false
  bpf:
    ctAnyMax: 262144
    ctTcpMax: 524288
    monitorAggregation: medium
    monitorFlags: all
    monitorInterval: 5s
    natMax: 524288
    preallocateMaps: false
    waitForMount: false
  cleanBpfState: false
  cleanState: false
  cluster:
    name: default
  cni:
    binPath: /opt/cni/bin
    chainingMode: none
    confFileMountPath: /tmp/cni-configuration
    confPath: /etc/cni/net.d
    configMapKey: cni-config
    customConf: false
    hostConfDirMountPath: /host/etc/cni/net.d
    install: true
  containerRuntime:
    integration: none
  daemon:
    runPath: /var/run/cilium
  debug:
    enabled: false
  enableXTSocketFallback: true
  encryption:
    enabled: false
    keyFile: keys
    mountPath: /etc/ipsec
    nodeEncryption: false
    secretName: cilium-ipsec-keys
  endpointRoutes:
    enabled: false
  eni: false
  etcd:
    clusterDomain: cluster.local
    clusterSize: 3
    enabled: false
    endpoints:
    - https://CHANGE-ME:2379
    managed: false
    ssl: false
  externalIPs:
    enabled: false
  flannel:
    enabled: false
    masterDevice: cni0
    uninstallOnExit: false
  hostPort:
    enabled: false
  hostServices:
    enabled: false
    protocols: tcp,udp
  hubble:
    enabled: false
    eventQueueSize: null
    flowBufferSize: null
    listenAddresses: []
    metricServer: null
    metrics: []
    socketPath: /var/run/cilium/hubble.sock
  identityAllocationMode: crd
  identityChangeGracePeriod: 5s
  installIptablesRules: true
  ipv4:
    enabled: true
  ipv6:
    enabled: false
  ipvlan:
    enabled: false
  k8s:
    requireIPv4PodCIDR: false
  kubeProxyReplacement: probe
  logSystemLoad: false
  masquerade: true
  nodePort:
    acceleration: none
    autoProtectPortRange: true
    enabled: false
    mode: hybrid
  nodeinit:
    bootstrapFile: /tmp/cilium-bootstrap-time
    enabled: false
  operatorPrometheus:
    enabled: false
    port: 6942
  policyAuditMode: false
  pprof:
    enabled: false
  prometheus:
    enabled: false
    port: 9090
    serviceMonitor:
      enabled: false
  psp:
    enabled: false
  pullPolicy: Always
  registry: docker.io/cilium
  remoteNodeIdentity: true
  sockops:
    enabled: false
  synchronizeK8sNodes: true
  tag: latest
  tls:
    secretsBackend: local
  tunnel: vxlan
  wellKnownIdentities:
    enabled: false
operator:
  enabled: true
  global:
    autoDirectNodeRoutes: false
    azure:
      enabled: false
    bpf:
      ctAnyMax: 262144
      ctTcpMax: 524288
      monitorAggregation: medium
      monitorFlags: all
      monitorInterval: 5s
      natMax: 524288
      preallocateMaps: false
      waitForMount: false
    cleanBpfState: false
    cleanState: false
    cluster:
      name: default
    cni:
      binPath: /opt/cni/bin
      chainingMode: none
      confFileMountPath: /tmp/cni-configuration
      confPath: /etc/cni/net.d
      configMapKey: cni-config
      customConf: false
      hostConfDirMountPath: /host/etc/cni/net.d
      install: true
    containerRuntime:
      integration: none
    daemon:
      runPath: /var/run/cilium
    debug:
      enabled: false
    enableXTSocketFallback: true
    encryption:
      enabled: false
      keyFile: keys
      mountPath: /etc/ipsec
      nodeEncryption: false
      secretName: cilium-ipsec-keys
    endpointRoutes:
      enabled: false
    eni: false
    etcd:
      clusterDomain: cluster.local
      clusterSize: 3
      enabled: false
      endpoints:
      - https://CHANGE-ME:2379
      managed: false
      ssl: false
    externalIPs:
      enabled: false
    flannel:
      enabled: false
      masterDevice: cni0
      uninstallOnExit: false
    hostPort:
      enabled: false
    hostServices:
      enabled: false
      protocols: tcp,udp
    hubble:
      enabled: false
      eventQueueSize: null
      flowBufferSize: null
      listenAddresses: []
      metricServer: null
      metrics: []
      socketPath: /var/run/cilium/hubble.sock
    identityAllocationMode: crd
    identityChangeGracePeriod: 5s
    installIptablesRules: true
    ipv4:
      enabled: true
    ipv6:
      enabled: false
    ipvlan:
      enabled: false
    k8s:
      requireIPv4PodCIDR: false
    kubeProxyReplacement: probe
    logSystemLoad: false
    masquerade: true
    nodePort:
      acceleration: none
      autoProtectPortRange: true
      enabled: false
      mode: hybrid
    nodeinit:
      bootstrapFile: /tmp/cilium-bootstrap-time
      enabled: false
    operatorPrometheus:
      enabled: false
      port: 6942
    policyAuditMode: false
    pprof:
      enabled: false
    prometheus:
      enabled: false
      port: 9090
      serviceMonitor:
        enabled: false
    psp:
      enabled: false
    pullPolicy: Always
    registry: docker.io/cilium
    remoteNodeIdentity: true
    sockops:
      enabled: false
    synchronizeK8sNodes: true
    tag: latest
    tls:
      secretsBackend: local
    tunnel: vxlan
    wellKnownIdentities:
      enabled: false
  image: operator
  resources: {}
  synchronizeK8sNodes: true
preflight:
  enabled: false
  tofqdnsPreCache: ""
  validateCNPs: true

HOOKS:
MANIFEST:
---
# Source: cilium/charts/agent/templates/serviceaccount.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: cilium
  namespace: kube-system
---
# Source: cilium/charts/operator/templates/serviceaccount.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: cilium-operator
  namespace: kube-system
---
# Source: cilium/charts/config/templates/configmap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: cilium-config
  namespace: kube-system
data:

  # Identity allocation mode selects how identities are shared between cilium
  # nodes by setting how they are stored. The options are "crd" or "kvstore".
  # - "crd" stores identities in kubernetes as CRDs (custom resource definition).
  #   These can be queried with:
  #     kubectl get ciliumid
  # - "kvstore" stores identities in a kvstore, etcd or consul, that is
  #   configured below. Cilium versions before 1.6 supported only the kvstore
  #   backend. Upgrades from these older cilium versions should continue using
  #   the kvstore by commenting out the identity-allocation-mode below, or
  #   setting it to "kvstore".
  identity-allocation-mode: crd

  # identity-change-grace-period is the grace period that needs to pass
  # before an endpoint that has changed its identity will start using
  # that new identity. During the grace period, the new identity has
  # already been allocated and other nodes in the cluster have a chance
  # to whitelist the new upcoming identity of the endpoint.
  identity-change-grace-period: "5s"

  # If you want to run cilium in debug mode change this value to true
  debug: "false"

  # Enable IPv4 addressing. If enabled, all endpoints are allocated an IPv4
  # address.
  enable-ipv4: "true"

  # Enable IPv6 addressing. If enabled, all endpoints are allocated an IPv6
  # address.
  enable-ipv6: "false"

  # If you want cilium monitor to aggregate tracing for packets, set this level
  # to "low", "medium", or "maximum". The higher the level, the less packets
  # that will be seen in monitor output.
  monitor-aggregation: medium

  # The monitor aggregation interval governs the typical time between monitor
  # notification events for each allowed connection.
  #
  # Only effective when monitor aggregation is set to "medium" or higher.
  monitor-aggregation-interval: 5s

  # The monitor aggregation flags determine which TCP flags which, upon the
  # first observation, cause monitor notifications to be generated.
  #
  # Only effective when monitor aggregation is set to "medium" or higher.
  monitor-aggregation-flags: all

  # bpf-ct-global-*-max specifies the maximum number of connections
  # supported across all endpoints, split by protocol: tcp or other. One pair
  # of maps uses these values for IPv4 connections, and another pair of maps
  # use these values for IPv6 connections.
  #
  # If these values are modified, then during the next Cilium startup the
  # tracking of ongoing connections may be disrupted. This may lead to brief
  # policy drops or a change in loadbalancing decisions for a connection.
  #
  # For users upgrading from Cilium 1.2 or earlier, to minimize disruption
  # during the upgrade process, set bpf-ct-global-tcp-max to 1000000.
  bpf-ct-global-tcp-max: "524288"
  bpf-ct-global-any-max: "262144"

  # bpf-nat-global-max specified the maximum number of entries in the BPF NAT
  # table.
  bpf-nat-global-max: "524288"

  # Pre-allocation of map entries allows per-packet latency to be reduced, at
  # the expense of up-front memory allocation for the entries in the maps. The
  # default value below will minimize memory usage in the default installation;
  # users who are sensitive to latency may consider setting this to "true".
  #
  # This option was introduced in Cilium 1.4. Cilium 1.3 and earlier ignore
  # this option and behave as though it is set to "true".
  #
  # If this value is modified, then during the next Cilium startup the restore
  # of existing endpoints and tracking of ongoing connections may be disrupted.
  # This may lead to policy drops or a change in loadbalancing decisions for a
  # connection for some time. Endpoints may need to be recreated to restore
  # connectivity.
  #
  # If this option is set to "false" during an upgrade from 1.3 or earlier to
  # 1.4 or later, then it may cause one-time disruptions during the upgrade.
  preallocate-bpf-maps: "false"

  # Regular expression matching compatible Istio sidecar istio-proxy
  # container image names
  sidecar-istio-proxy-image: "cilium/istio_proxy"

  # Encapsulation mode for communication between nodes
  # Possible values:
  #   - disabled
  #   - vxlan (default)
  #   - geneve
  tunnel: vxlan

  # Name of the cluster. Only relevant when building a mesh of clusters.
  cluster-name: default

  # DNS Polling periodically issues a DNS lookup for each `matchName` from
  # cilium-agent. The result is used to regenerate endpoint policy.
  # DNS lookups are repeated with an interval of 5 seconds, and are made for
  # A(IPv4) and AAAA(IPv6) addresses. Should a lookup fail, the most recent IP
  # data is used instead. An IP change will trigger a regeneration of the Cilium
  # policy for each endpoint and increment the per cilium-agent policy
  # repository revision.
  #
  # This option is disabled by default starting from version 1.4.x in favor
  # of a more powerful DNS proxy-based implementation, see [0] for details.
  # Enable this option if you want to use FQDN policies but do not want to use
  # the DNS proxy.
  #
  # To ease upgrade, users may opt to set this option to "true".
  # Otherwise please refer to the Upgrade Guide [1] which explains how to
  # prepare policy rules for upgrade.
  #
  # [0] http://docs.cilium.io/en/stable/policy/language/#dns-based
  # [1] http://docs.cilium.io/en/stable/install/upgrade/#changes-that-may-require-action
  tofqdns-enable-poller: "false"

  # wait-bpf-mount makes init container wait until bpf filesystem is mounted
  wait-bpf-mount: "false"

  masquerade: "true"
  enable-xt-socket-fallback: "true"
  install-iptables-rules: "true"
  auto-direct-node-routes: "false"
  kube-proxy-replacement:  "probe"
  enable-host-reachable-services: "false"
  enable-host-port: "false"
  enable-external-ips: "false"
  enable-node-port: "false"
  node-port-mode: "hybrid"
  node-port-acceleration: "none"
  enable-auto-protect-node-port-range: "true"
  # Chaining mode is set to portmap, enable health checking
  enable-endpoint-health-checking: "true"
  enable-well-known-identities: "false"
  enable-remote-node-identity: "true"

  synchronize-k8s-nodes: "true"
  policy-audit-mode: "false"
  operator-api-serve-addr: '127.0.0.1:9234'


  # A space separated list of iptables chains to disable when installing feeder rules.
  disable-iptables-feeder-rules: ""
---
# Source: cilium/charts/agent/templates/clusterrole.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: cilium
rules:
- apiGroups:
  - networking.k8s.io
  resources:
  - networkpolicies
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - discovery.k8s.io
  resources:
  - endpointslices
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - ""
  resources:
  - namespaces
  - services
  - nodes
  - endpoints
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - ""
  resources:
  - pods
  - nodes
  verbs:
  - get
  - list
  - watch
  - update
- apiGroups:
  - ""
  resources:
  - nodes
  - nodes/status
  verbs:
  - patch
- apiGroups:
  - apiextensions.k8s.io
  resources:
  - customresourcedefinitions
  verbs:
  - create
  - get
  - list
  - watch
  - update
- apiGroups:
  - cilium.io
  resources:
  - ciliumnetworkpolicies
  - ciliumnetworkpolicies/status
  - ciliumclusterwidenetworkpolicies
  - ciliumclusterwidenetworkpolicies/status
  - ciliumendpoints
  - ciliumendpoints/status
  - ciliumnodes
  - ciliumnodes/status
  - ciliumidentities
  - ciliumidentities/status
  verbs:
  - '*'
---
# Source: cilium/charts/operator/templates/clusterrole.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: cilium-operator
rules:
- apiGroups:
  - ""
  resources:
  # to automatically delete [core|kube]dns pods so that are starting to being
  # managed by Cilium
  - pods
  verbs:
  - get
  - list
  - watch
  - delete
- apiGroups:
  - discovery.k8s.io
  resources:
  - endpointslices
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - ""
  resources:
  # to automatically read from k8s and import the node's pod CIDR to cilium's
  # etcd so all nodes know how to reach another pod running in in a different
  # node.
  - nodes
  # to perform the translation of a CNP that contains `ToGroup` to its endpoints
  - services
  - endpoints
  # to check apiserver connectivity
  - namespaces
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - cilium.io
  resources:
  - ciliumnetworkpolicies
  - ciliumnetworkpolicies/status
  - ciliumclusterwidenetworkpolicies
  - ciliumclusterwidenetworkpolicies/status
  - ciliumendpoints
  - ciliumendpoints/status
  - ciliumnodes
  - ciliumnodes/status
  - ciliumidentities
  - ciliumidentities/status
  verbs:
  - '*'
---
# Source: cilium/charts/agent/templates/clusterrolebinding.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: cilium
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: cilium
subjects:
- kind: ServiceAccount
  name: cilium
  namespace: kube-system
---
# Source: cilium/charts/operator/templates/clusterrolebinding.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: cilium-operator
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: cilium-operator
subjects:
- kind: ServiceAccount
  name: cilium-operator
  namespace: kube-system
---
# Source: cilium/charts/agent/templates/daemonset.yaml
apiVersion: apps/v1
kind: DaemonSet
metadata:
  labels:
    k8s-app: cilium
  name: cilium
  namespace: kube-system
spec:
  selector:
    matchLabels:
      k8s-app: cilium
  template:
    metadata:
      annotations:
        # This annotation plus the CriticalAddonsOnly toleration makes
        # cilium to be a critical pod in the cluster, which ensures cilium
        # gets priority scheduling.
        # https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/
        scheduler.alpha.kubernetes.io/critical-pod: ""
      labels:
        k8s-app: cilium
    spec:
      containers:
      - args:
        - --config-dir=/tmp/cilium/config-map
        command:
        - cilium-agent
        livenessProbe:
          exec:
            command:
            - cilium
            - status
            - --brief
          failureThreshold: 10
          # The initial delay for the liveness probe is intentionally large to
          # avoid an endless kill & restart cycle if in the event that the initial
          # bootstrapping takes longer than expected.
          initialDelaySeconds: 120
          periodSeconds: 30
          successThreshold: 1
          timeoutSeconds: 5
        readinessProbe:
          exec:
            command:
            - cilium
            - status
            - --brief
          failureThreshold: 3
          initialDelaySeconds: 5
          periodSeconds: 30
          successThreshold: 1
          timeoutSeconds: 5
        env:
        - name: K8S_NODE_NAME
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: spec.nodeName
        - name: CILIUM_K8S_NAMESPACE
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
        - name: CILIUM_FLANNEL_MASTER_DEVICE
          valueFrom:
            configMapKeyRef:
              key: flannel-master-device
              name: cilium-config
              optional: true
        - name: CILIUM_FLANNEL_UNINSTALL_ON_EXIT
          valueFrom:
            configMapKeyRef:
              key: flannel-uninstall-on-exit
              name: cilium-config
              optional: true
        - name: CILIUM_CLUSTERMESH_CONFIG
          value: /var/lib/cilium/clustermesh/
        - name: CILIUM_CNI_CHAINING_MODE
          valueFrom:
            configMapKeyRef:
              key: cni-chaining-mode
              name: cilium-config
              optional: true
        - name: CILIUM_CUSTOM_CNI_CONF
          valueFrom:
            configMapKeyRef:
              key: custom-cni-conf
              name: cilium-config
              optional: true
        image: "docker.io/cilium/cilium:latest"
        imagePullPolicy: Always
        lifecycle:
          postStart:
            exec:
              command:
              - "/cni-install.sh"
              - "--enable-debug=false"
          preStop:
            exec:
              command:
              - /cni-uninstall.sh
        name: cilium-agent
        securityContext:
          capabilities:
            add:
            - NET_ADMIN
            - SYS_MODULE
          privileged: true
        volumeMounts:
        - mountPath: /sys/fs/bpf
          name: bpf-maps
        - mountPath: /var/run/cilium
          name: cilium-run
        - mountPath: /host/opt/cni/bin
          name: cni-path
        - mountPath: /host/etc/cni/net.d
          name: etc-cni-netd
        - mountPath: /var/lib/cilium/clustermesh
          name: clustermesh-secrets
          readOnly: true
        - mountPath: /tmp/cilium/config-map
          name: cilium-config-path
          readOnly: true
          # Needed to be able to load kernel modules
        - mountPath: /lib/modules
          name: lib-modules
          readOnly: true
        - mountPath: /run/xtables.lock
          name: xtables-lock
      hostNetwork: true
      initContainers:
      - command:
        - /init-container.sh
        env:
        - name: CILIUM_ALL_STATE
          valueFrom:
            configMapKeyRef:
              key: clean-cilium-state
              name: cilium-config
              optional: true
        - name: CILIUM_BPF_STATE
          valueFrom:
            configMapKeyRef:
              key: clean-cilium-bpf-state
              name: cilium-config
              optional: true
        - name: CILIUM_WAIT_BPF_MOUNT
          valueFrom:
            configMapKeyRef:
              key: wait-bpf-mount
              name: cilium-config
              optional: true
        image: "docker.io/cilium/cilium:latest"
        imagePullPolicy: Always
        name: clean-cilium-state
        securityContext:
          capabilities:
            add:
            - NET_ADMIN
          privileged: true
        volumeMounts:
        - mountPath: /sys/fs/bpf
          name: bpf-maps
          mountPropagation: HostToContainer
        - mountPath: /var/run/cilium
          name: cilium-run
        resources:
          requests:
            cpu: 100m
            memory: 100Mi
      restartPolicy: Always
      priorityClassName: system-node-critical
      serviceAccount: cilium
      serviceAccountName: cilium
      terminationGracePeriodSeconds: 1
      tolerations:
      - operator: Exists
      volumes:
        # To keep state between restarts / upgrades
      - hostPath:
          path: /var/run/cilium
          type: DirectoryOrCreate
        name: cilium-run
        # To keep state between restarts / upgrades for bpf maps
      - hostPath:
          path: /sys/fs/bpf
          type: DirectoryOrCreate
        name: bpf-maps
      # To install cilium cni plugin in the host
      - hostPath:
          path:  /opt/cni/bin
          type: DirectoryOrCreate
        name: cni-path
        # To install cilium cni configuration in the host
      - hostPath:
          path: /etc/cni/net.d
          type: DirectoryOrCreate
        name: etc-cni-netd
        # To be able to load kernel modules
      - hostPath:
          path: /lib/modules
        name: lib-modules
        # To access iptables concurrently with other processes (e.g. kube-proxy)
      - hostPath:
          path: /run/xtables.lock
          type: FileOrCreate
        name: xtables-lock
        # To read the clustermesh configuration
      - name: clustermesh-secrets
        secret:
          defaultMode: 420
          optional: true
          secretName: cilium-clustermesh
        # To read the configuration from the config map
      - configMap:
          name: cilium-config
        name: cilium-config-path
  updateStrategy:
    rollingUpdate:
      maxUnavailable: 2
    type: RollingUpdate
---
# Source: cilium/charts/operator/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    io.cilium/app: operator
    name: cilium-operator
  name: cilium-operator
  namespace: kube-system
spec:
  replicas: 1
  selector:
    matchLabels:
      io.cilium/app: operator
      name: cilium-operator
  strategy:
    rollingUpdate:
      maxSurge: 1
      maxUnavailable: 1
    type: RollingUpdate
  template:
    metadata:
      annotations:
      labels:
        io.cilium/app: operator
        name: cilium-operator
    spec:
      containers:
      - args:
        - --config-dir=/tmp/cilium/config-map
        - --debug=$(CILIUM_DEBUG)
        command:
        - cilium-operator
        env:
        - name: K8S_NODE_NAME
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: spec.nodeName
        - name: CILIUM_K8S_NAMESPACE
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
        - name: CILIUM_DEBUG
          valueFrom:
            configMapKeyRef:
              key: debug
              name: cilium-config
              optional: true
        - name: AWS_ACCESS_KEY_ID
          valueFrom:
            secretKeyRef:
              key: AWS_ACCESS_KEY_ID
              name: cilium-aws
              optional: true
        - name: AWS_SECRET_ACCESS_KEY
          valueFrom:
            secretKeyRef:
              key: AWS_SECRET_ACCESS_KEY
              name: cilium-aws
              optional: true
        - name: AWS_DEFAULT_REGION
          valueFrom:
            secretKeyRef:
              key: AWS_DEFAULT_REGION
              name: cilium-aws
              optional: true
        image: "docker.io/cilium/operator:latest"
        imagePullPolicy: Always
        name: cilium-operator
        livenessProbe:
          httpGet:
            host: '127.0.0.1'
            path: /healthz
            port: 9234
            scheme: HTTP
          initialDelaySeconds: 60
          periodSeconds: 10
          timeoutSeconds: 3
        volumeMounts:
        - mountPath: /tmp/cilium/config-map
          name: cilium-config-path
          readOnly: true
      hostNetwork: true
      restartPolicy: Always
      priorityClassName: system-cluster-critical
      serviceAccount: cilium-operator
      serviceAccountName: cilium-operator
      volumes:
        # To read the configuration from the config map
      - configMap:
          name: cilium-config
        name: cilium-config-path

NOTES:
You have successfully installed Cilium

Your release version is 1.7.90

For any further help, visit https://docs.cilium.io/en/v1.7/gettinghelp

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

File: install/kubernetes/cilium/templates/NOTES.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+You have successfully installed Cilium {{ title .Chart.Name }}
+
+Your release version is {{ .Chart.Version }}
```
Comment (joestringer) [inline install/kubernetes/cilium/templates/NOTES.txt]: nit:

These are still regular sentences, so they should probably have punctuation right?

```suggestion
You have successfully installed Cilium {{ title .Chart.Name }}.

Your release version is {{ .Chart.Version }}.
```

Comment (soumynathan) [reply install/kubernetes/cilium/templates/NOTES.txt]: ok will address your comments.

Comment (soumynathan) [reply install/kubernetes/cilium/templates/NOTES.txt]: Done

**Comment by joestringer** [comment]
Hit #10977 in Travis arm64 test, we can safely ignore this.

File: install/kubernetes/cilium/templates/NOTES.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+You have successfully installed Cilium {{ title .Chart.Name }}.
```
Comment (joestringer) [inline install/kubernetes/cilium/templates/NOTES.txt]: This prints:

```
You have successfully installed Cilium Cilium.

Your release version is 1.7.90.

For any further help, visit https://docs.cilium.io/en/v1.7/gettinghelp
```

Please avoid printing Cilium twice here.

**Review by joestringer — CHANGES_REQUESTED** [review]

File: install/kubernetes/cilium/templates/NOTES.txt (hunk 2 above)
Comment (soumynathan) [reply install/kubernetes/cilium/templates/NOTES.txt]: Mybad, will fix it.

**Review by joestringer — APPROVED** [review]

**Pull request merged**